### ISSUE TITLE
Bug: Media Switch Direction not properly detected

### ISSUE THREAD

**Issue by jeengbe** [ModernFlyouts-Community/ModernFlyouts#496]
**Describe the bug:**
When listening to a playlist or liked songs on Spotify, skipping and switching to the previous track, the direction the new songs come flying in is seemingly random. This issue does not happen with Groove Music or YouTube.

**To Reproduce:**
Steps to reproduce the behaviour:
1. Open Spotify
2. Queue any playlist
3. Skip a couple of songs, observe

**Videos:**

https://user-images.githubusercontent.com/51749023/110831711-ce246c00-829a-11eb-9009-91a07bcf8b3e.mp4

**OS Version:**
Windows 10 Pro 10.0.19042 Build 19042

**ModernFlyouts Version:**
0.9.0.0

**Additional context:**
Tested with Shuffle, Repeat on/off. Issue seems to only occur with Spotify

**Comment by Samuel12321** [comment]
Sounds like a Spotify issue to me - @ShankarBUS any thoughts

@jeengbe does this occur with the windows default flyout?

**Comment by Cyberdroid1** [comment]
> @jeengbe does this occur with the windows default flyout?

Windows one doesn't have animations lol

**Comment by Samuel12321** [comment]
oh right, i misinterpreted that, i thought he meant it was skipping to the wrong song, not that the animations weren't going the right way.

**Comment by jeengbe** [comment]
I believe this is a design issue. A song's track number is determined by its position on the album, not in the current playlist, thus giving arbitrary looking results. The issue does not occur when playing songs from the same album only. Without shuffle, it works perfectly fine, though, enabling shuffle (expectedly) renders the same result.
https://github.com/ModernFlyouts-Community/ModernFlyouts/blob/104025668a2778334c859051afe97875f999ae5b/ModernFlyouts.Core/Media/Control/NowPlayingMediaSession.cs#L179-L184

Shouldn't the code cause an enormous number of exceptions for two succeeding songs which are both the first in their albums?

**Comment by ShankarBUS** [comment]
> I believe this is a design issue.

Yeah, I am aware of this issue since I started adding this feature. This is just a blind assumption method. Not accurate.
But only Spotify produces this problem. Groove Music works well.

> A song's track number is determined by its position on the album, not in the current playlist, thus giving arbitrary looking results.

I am not sure about that. But AFAIK, it must be the position of the song in a playlist. I will verify this.

> The issue does not occur when playing songs from the same album only. Without shuffle, it works perfectly fine, though, enabling shuffle (expectedly) renders the same result.

Yeah shuffling doesn't update the track number. Groove Music however works perfectly with this.

> Shouldn't the code cause an enormous number of exceptions for two succeeding songs which are both the first in their albums?

![](https://media.giphy.com/media/U1XhGr8CWqvVC/giphy.gif)

I will do something about this ASAP.

**Comment by ShankarBUS** [comment]
@jeengbe Thank you so much mate!

**Comment by jeengbe** [comment]
Oh, absolutely no thing, thank you for the beautiful application sir! :D
